netket/hilbert/abstract_continuous_space_particle.py

```diff
@@ -0,0 +1,78 @@
+from typing import Tuple
+
+import jax.numpy as jnp
+
+from .abstract_hilbert import AbstractHilbert
+
+
+class AbstractParticle(AbstractHilbert):
+    """Abstract class for the Hilbert space of particles
+    in continuous space.
+    This class defines the common interface that
+    can be used to interact with particles defined
+    in continuous space.
+    """
+
+    def __init__(self, N: int, L: Tuple[float, ...], pbc: Tuple[bool, ...]):
+        """
+        Constructs new ``Particles`` given specifications
+         of the continuous space they are defined in.
+
+        Args:
+            N: Number of particles
+            L: spatial extension in each spatial dimension
+            pbc: Whether or not to use periodic boundary
+                conditions for each spatial dimension
+        """
+        self._N = N
+        self._L = L
+        self._pbc = pbc
+
+        if not len(self._L) == len(self._pbc):
+            raise AssertionError(
+                """You need to define boundary conditions for each spatial dimension."""
```

Comment:
AThat's a ValueError. And with the change above we can rephrase the error string
```suggestion
            raise ValueError(
                """`pbc` must be either a bool or a tuple indicating the periodicity of each spatial dimension."""
```